### PR TITLE
Feature: Add Create group page

### LINKED ISSUE #116: Add Create group page
Add the creation/deletion of a group, with a mandatory user and role

image

### PR DESCRIPTION
**Pull request by justinehell**
Resolves #116

### CODE REVIEW

File: src/components/tables/RolesTable.vue [hunk 1]
```diff
@@ -116,6 +120,18 @@ const props = defineProps({
     type: Boolean,
     default: true,
   },
+  showFiltersAndPagination: {
+    type: Boolean,
+    default: true,
+  },
+  noData: {
```
Comment (Zorin95670) [inline src/components/tables/RolesTable.vue:130]: noDataText

File: src/components/tables/RolesTable.vue [hunk 2]
```diff
@@ -116,6 +120,18 @@ const props = defineProps({
...
+    type: String,
+    default: 'noData',
```
Comment (Zorin95670) [inline src/components/tables/RolesTable.vue:132]: default null

File: src/components/tables/RolesTable.vue [hunk 3]
```diff
@@ -116,6 +120,18 @@ const props = defineProps({
...
+  },
+  noDataIcon: {
+    type: String,
+    default: 'noData',
```
Comment (Zorin95670) [inline src/components/tables/RolesTable.vue:136]: default null

File: src/components/tables/RolesTable.vue [hunk 4]
```diff
@@ -68,11 +71,12 @@
       <template #no-data>
         <div class="full-width row flex-center q-gutter-sm">
           <q-icon
+            v-if="noDataIcon"
             size="2em"
-            :name="$t('RolesTable.icon.noData')"
+            :name="$t(`RolesTable.icon.${noDataIcon}`)"
```
Comment (Zorin95670) [inline src/components/tables/RolesTable.vue:79]: :name="noDataIcon"

File: src/components/tables/RolesTable.vue [hunk 5]
```diff
@@ -68,11 +71,12 @@
...
           />
           <span>
```
Comment (Zorin95670) [inline src/components/tables/RolesTable.vue:81]: v-if="noDataLabel"

File: src/components/tables/RolesTable.vue [hunk 6]
```diff
@@ -116,6 +120,18 @@ const props = defineProps({
     type: Boolean,
     default: true,
   },
+  showFiltersAndPagination: {
```
Comment (Zorin95670) [inline src/components/tables/RolesTable.vue:126]: divide in 2 (filters and pagination)

File: src/i18n/en-US/index.js [hunk 7]
```diff
@@ -182,6 +182,7 @@ export default {
       removeAction: 'Remove user',
       detachAction: 'Detach user',
       noData: 'No users found.',
+      createGroupMessage: 'Please add at least one member to create a group',
```
Comment (Zorin95670) [inline src/i18n/en-US/index.js:189]: move into AddGroupPage

File: src/i18n/en-US/index.js [hunk 8]
```diff
@@ -198,6 +199,7 @@ export default {
       removeAction: 'Remove role',
       detachAction: 'Detach role',
       noData: 'No roles found.',
+      createGroupMessage: 'Please add at least one role to create a group',
```
Comment (Zorin95670) [inline src/i18n/en-US/index.js:206]: delete it

File: src/pages/AddGroupPage.vue [hunk 9]
```diff
@@ -0,0 +1,302 @@
...
+          :label="$t('AddGroupPage.text.attachRole')"
+          :icon="$t('AddGroupPage.icon.attach')"
+          @click="openAttachRoleToGroupDialog"
+        />
+        <roles-table
+          :roles="selectedRoles"
+          :show-action="false"
+          :detach-action="false"
+          :show-filters-and-pagination="false"
+          no-data="createGroupMessage"
```
Comment (Zorin95670) [inline src/pages/AddGroupPage.vue:86]: :no-data-label="$t('AddGroupPage.text.(...))"

File: src/components/tables/RolesTable.vue (hunk 1 above)
Comment (justinehell) [reply src/components/tables/RolesTable.vue:130]: d

File: src/components/tables/RolesTable.vue (hunk 3 above)
Comment (justinehell) [reply src/components/tables/RolesTable.vue:136]: d

File: src/components/tables/RolesTable.vue (hunk 4 above)
Comment (justinehell) [reply src/components/tables/RolesTable.vue:79]: d

File: src/components/tables/RolesTable.vue (hunk 5 above)
Comment (justinehell) [reply src/components/tables/RolesTable.vue:81]: d

File: src/components/tables/RolesTable.vue [hunk 10]
```diff
@@ -68,11 +71,12 @@
...
         <div class="full-width row flex-center q-gutter-sm">
           <q-icon
+            v-if="noDataIcon"
             size="2em"
-            :name="$t('RolesTable.icon.noData')"
+            :name="$t(`RolesTable.icon.${noDataIcon}`)"
           />
           <span>
-            {{ $t('RolesTable.text.noData') }}
+            {{ $t(`RolesTable.text.${noData}`) }}
```
[the remark this answers is not in the document]
Comment (justinehell) [reply src/components/tables/RolesTable.vue:82]: d

File: src/components/tables/RolesTable.vue (hunk 6 above)
Comment (justinehell) [reply src/components/tables/RolesTable.vue:126]: d

File: src/pages/AddGroupPage.vue (hunk 9 above)
Comment (justinehell) [reply src/pages/AddGroupPage.vue:86]: d

File: src/i18n/en-US/index.js (hunk 7 above)
Comment (justinehell) [reply src/i18n/en-US/index.js:189]: d

File: src/pages/AddGroupPage.vue [hunk 11]
```diff
@@ -0,0 +1,301 @@
...
+          :show-pagination="false"
+          :no-data-label="$t('AddGroupPage.text.addRoletoGroupMessage')"
+          class="full-width"
+          @remove="updateSelectedRoles"
+        />
+      </div>
+      <q-space />
+      <div class="full-width row justify-center">
+        <q-btn
+          label="cancel"
```
Comment (Zorin95670) [inline src/pages/AddGroupPage.vue:94]: translation

File: src/pages/AddGroupPage.vue [hunk 12]
```diff
@@ -0,0 +1,301 @@
...
+          color="negative"
+          class="q-mr-xl"
+          type="reset"
+          data-cy="group_button_cancel"
+        />
+        <q-btn
+          label="Add"
```
Comment (Zorin95670) [inline src/pages/AddGroupPage.vue:101]: translation

File: src/pages/AddGroupPage.vue (hunk 11 above)
Comment (justinehell) [reply src/pages/AddGroupPage.vue:94]: done new commit

File: src/pages/AddGroupPage.vue (hunk 12 above)
Comment (justinehell) [reply src/pages/AddGroupPage.vue:101]: done

File: src/pages/AddGroupPage.vue [hunk 13]
```diff
@@ -0,0 +1,301 @@
...
+ */
+async function setSelectedRoles(event) {
+  selectedRoles.value = event;
+}
+
+/**
+ * Update selected users.
+ * @param {object} userToRemove - User to remove from selected users.
+ */
+async function updateSelectedUsers(userToRemove) {
```
Comment (Zorin95670) [inline src/pages/AddGroupPage.vue:166]: rename to removeUser(user)

File: src/pages/AddGroupPage.vue [hunk 14]
```diff
@@ -0,0 +1,301 @@
...
+  selectedUsers.value = selectedUsers.value.filter(({ login }) => login !== userToRemove.login);
+}
+
+/**
+ * Update selected roles.
+ * @param {object} roleToRemove - Role to remove from selected roles.
+ */
+async function updateSelectedRoles(roleToRemove) {
```
Comment (Zorin95670) [inline src/pages/AddGroupPage.vue:174]: rename to removeRole(role)

File: src/pages/AddGroupPage.vue (hunk 13 above)
Comment (justinehell) [reply src/pages/AddGroupPage.vue]: done

File: src/pages/AddGroupPage.vue (hunk 14 above)
Comment (justinehell) [reply src/pages/AddGroupPage.vue]: done

**Review by Zorin95670 — APPROVED** [review]

**Pull request merged**